### PR TITLE
AEA-1121 Filling coverage gaps for `helpers/search/model`; Resolve semantics

### PR DESCRIPTION
**Pull request by 5A11**
This PR fills coverage holes in `helpers.search.model`.
It also corrects the semantics of <, <=, >, >=, relations in ConstraintTypes.

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that code coverage does not decrease.
- [ ] I have checked that the documentation about the `aea cli` tool works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: tests/test_helpers/test_search/test_models.py [hunk 1]
```diff
@@ -167,67 +169,164 @@ def test_description():
...
     """Test ConstraintType."""
-    for cons_type in [
-        ConstraintTypes.EQUAL,
-        ConstraintTypes.NOT_EQUAL,
+    constraint_type_values = {
+        "int": 12,
+        "bool": True,
+        "float": 10.4,
+        "str": "some_string",
+        "location": Location(1.1, 2.2),
```
Comment (DavidMinarsch) [inline tests/test_helpers/test_search/test_models.py:177]: Please have some tests which explicitly use the keywords `latitude` and `longitude` and assert encoding/decoding leads to correct recovery!

Comment (5A11) [reply tests/test_helpers/test_search/test_models.py:179]: Done

**Review by DavidMinarsch — APPROVED** [review]
LGTM

**Review by solarw — APPROVED** [review]
LGTM

**Pull request merged**